coalib/bearlib/languages/documentation/DocBaseClass.py

```diff
@@ -43,9 +44,17 @@ def extract(content, language, docstyle):
                                    DocumentationComment found in the content.
         """
         docstyle_definition = DocstyleDefinition.load(language, docstyle)
-        return extract_documentation_with_markers(
+        doc_comment_iterator = extract_documentation_with_markers(
                         content, docstyle_definition)
 
+        doc_comment_iterator = self.instantiate_padding(
+            content, doc_comment_iterator)
+
+        doc_comment_iterator = self.instantiate_kind(
+            content, doc_comment_iterator)
```

Comment:
You have all you need at the end of [extract_documentation_with_markers](https://github.com/damngamerz/coala/blob/blankline/coalib/bearlib/languages/documentation/DocumentationExtraction.py#L253). I see no reason, why we should return an incomplete DocComment there and then use two more methods to finalize it.